mteb/abstasks/AbsTaskReranking.py

```diff
@@ -1,98 +1,216 @@
+
+logger = logging.getLogger(__name__)
+
+OLD_FORMAT_RERANKING_TASKS = [
+    "AskUbuntuDupQuestions",
+    "MindSmallReranking",
+    "SciDocsRR",
+    "StackOverflowDupQuestions",
+    "WebLINXCandidatesReranking",
+    "AlloprofReranking",
+    "SyntecReranking",
+    "VoyageMMarcoReranking",
+    "ESCIReranking",
+    "MIRACLReranking",
+    "WikipediaRerankingMultilingual",
+    "RuBQReranking",
+    "T2Reranking",
+    "MMarcoReranking",
+    "CMedQAv1-reranking",
+    "CMedQAv2-reranking",
+]
+
+
+class AbsTaskReranking(AbsTaskRetrieval):
+    """Abstract class for re-ranking experiments. This is mostly the same as the RetrievalEvaluator, but as previously it wasn't we need to keep it to transform old dataset versions into the same format.
+
+    New Format:
+    -----------
+    Same as AbsTaskRetrieval, but with a top_ranked file that contains the passages to rerank. The dataset should contain the following columns:
+
+        self.corpus: dict[str, dict[str, str]]
+            Semantically, it should contain dict[split_name, dict[sample_id, dict[str, str]]]
+            E.g. {"test": {"document_one": {"_id": "d1", "title": "title", "text": "text"}}}
+
+        self.queries: dict[str, dict[str, Union[str, list[str]]]]
+            Semantically, it should contain dict[split_name, dict[sample_id, str]] or dict[split_name, dict[sample_id, list[str]]] for conversations
+            E.g. {"test": {"q1": "query"}}
+            or {"test": {"q1": ["turn1", "turn2", "turn3"]}}
+
+        self.relevant_docs: dict[str, dict[str, dict[str, int]]]
+            Semantically, it should contain dict[split_name, dict[sample_id, dict[doc_id, score]]]
+            E.g.: {"test": {"q1": {"document_one": 1}}}
+
+        self.top_ranked: dict[str, dict[str, list[str]]] or dict[str, dict[str, dict[str, float]]]
+            Semantically, it should contain dict[split_name, dict[sample_id, list[doc_id]]] or dict[split_name, dict[sample_id, dict[doc_id, score]]]
+            E.g.: {"test": {"q1": ["document_one", "document_two"]}} or {"test": {"q1": {"document_one": 1, "document_two": 0.5}}}
+
+    Old Format:
+    -----------
     self.load_data() must generate a huggingface dataset with a split matching self.metadata_dict["eval_splits"], and assign it to self.dataset. It must contain the following columns:
         query: str
         positive: list[str]
         negative: list[str]
     """
 
     def __init__(self, **kwargs):
-        super().__init__(**kwargs)
+        super(AbsTaskRetrieval, self).__init__(**kwargs)
+
+    def load_data(self, **kwargs):
+        if self.data_loaded:
+            return
+
+        if self.metadata.name in OLD_FORMAT_RERANKING_TASKS:
+            self.dataset = datasets.load_dataset(**self.metadata_dict["dataset"])  # type: ignore
+            self.dataset_transform()
+        else:
+            # use AbsTaskRetrieval default to load the data
+            # TODO: need to make sure top_ranked comes back
```

Comment:
?

Reply:
Sorry, left this as a note to myself but it brings up a good question for you. We currently have `HFDataLoader` which loads the Retrieval data.  But we could combine it with the InstructionHFDataLoader and also have it do the reranking loading.  It would just need to have options to (1) load an instruction config and (2) load a top_ranked config for reranking. 

Does that seem okay to you? It get a bit more complex.  I could also extend and make a new class, keeping as much as possible the same between them

Reply:
I was thinking something like checking if the dataset has a config called "instructions" or "top_ranked" and if it does to load them. Agree we don't want to use `trust_remote_code`!

Reply:
Sounds reasonable